tools/cpp/unix_cc_configure.bzl

```diff
@@ -208,6 +208,8 @@ def _find_linker_path(repository_ctx, cc, linker, is_clang):
         for flag in line.split(" "):
```

Comment:
t seems this code has a lot of legacy cruft that we can clean up.  IIUC this only needs to support clang now, and we are just looking for the linker path, right?  So instead of looping through the entire commandline can we just check the first arg to see if it matches the linker (with possibly a space before it)?

Reply:
unix_cc_configure does support GCC as well, it's just on darwin that it will realistically only ever be clang. Related: https://github.com/bazelbuild/bazel/commit/ef3f05852b3bf542903f091184da96b96354a13d

Reply:
But I think this code fragment needs to only support clang, due to line 202, from https://github.com/bazelbuild/bazel/pull/12729.

Reply:
More to the point, this code evolved from back when the code needed to look for "-fuse-ld=<linker>" arg for gcc, but the linker binary for clang.  That was why it iterated through the whole commandline.  Now that it is only looking for the linker binary for clang, it should only need to look at the first arg.

Reply:
good point, I changed the parsing to extract the first argument, from the last line, of the output. For reference here's the output I see on macOS:

```
% clang "tools/cpp/empty.cc" -o /dev/null -fuse-ld=lld -v
Apple clang version 14.0.0 (clang-1400.0.29.102)
Target: arm64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 "/Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 ...
clang -cc1 version 14.0.0 (clang-1400.0.29.102) default target arm64-apple-darwin21.6.0
ignoring nonexistent directory "/usr/local/include"
ignoring nonexistent directory "/Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1
 /Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include
 /Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
 /Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
 "/Users/ksmiley/bin/ld64.lld" -demangle -no_deduplicate -dynamic -arch arm64 -platform_version macos 12.0.0 12.3 -syslibroot /Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -o /dev/null -L/usr/local/lib /var/folders/gj/wf3swl0x215b2sq1qy84kzkm0000gn/T/empty-ff40e4.o -lSystem /Applications/Xcode-14.0.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.osx.a
```